File: test/test_unit_numeric_array.rb
require_relative "../lib/rgfa.rb"
require "test/unit"

TestUnit ||= Module.new

# note: API public methods are tested in test_api_tags
class TestUnit::NumericArray < Test::Unit::TestCase

  def test_integer_type
    v = {}
    [8,16,32,64,128].each do |b|
      v[b] = 1 << (b/2)
    end
    assert_equal("C", RGFA::NumericArray.integer_type(0..v[8]))
    assert_equal("c", RGFA::NumericArray.integer_type(-1..v[8]))
    assert_equal("S", RGFA::NumericArray.integer_type(0..v[16]))
    assert_equal("s", RGFA::NumericArray.integer_type(-1..v[16]))
    assert_equal("I", RGFA::NumericArray.integer_type(0..v[32]))
    assert_equal("i", RGFA::NumericArray.integer_type(-1..v[32]))
    assert_raise(RGFA::ValueError) {RGFA::NumericArray.integer_type(0..v[64])}
    assert_raise(RGFA::ValueError) {RGFA::NumericArray.integer_type(-1..v[64])}
    assert_raise(RGFA::ValueError) {RGFA::NumericArray.integer_type(0..v[128])}
    assert_raise(RGFA::ValueError) {RGFA::NumericArray.integer_type(-1..v[128])}
  end

end

File: tests/test_unit_numeric_array.py
import unittest
import gfapy

class TestUnitNumericArray(unittest.TestCase):

  def test_integer_type(self):
    v = {b: 2**(b/2) for b in [8,16,32,64,128]}
    self.assertEqual("C", gfapy.NumericArray.integer_type((0,v[8])))
    self.assertEqual("c", gfapy.NumericArray.integer_type((-1,v[8])))
    self.assertEqual("S", gfapy.NumericArray.integer_type((0,v[16])))
    self.assertEqual("s", gfapy.NumericArray.integer_type((-1,v[16])))
    self.assertEqual("I", gfapy.NumericArray.integer_type((0,v[32])))
    self.assertEqual("i", gfapy.NumericArray.integer_type((-1,v[32])))
    self.assertRaises(gfapy.ValueError,
        gfapy.NumericArray.integer_type, (0,v[64]))
    self.assertRaises(gfapy.ValueError,
        gfapy.NumericArray.integer_type, (-1,v[64]))
    self.assertRaises(gfapy.ValueError,
        gfapy.NumericArray.integer_type, (0,v[128]))
    self.assertRaises(gfapy.ValueError,
        gfapy.NumericArray.integer_type, (-1,v[128]))
